modules/ossm-install-cert-manager.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v2x/ossm-security.adoc
+
+:_content-type: PROCEDURE
+[id="ossm-install-cert-manager_{context}"]
+== Install cert-manager
+
+.Procedure
+
+. Create root cluster issuer:
++
+[source, terminal]
+----
+oc apply -f cluster-issuer.yaml
+----
++
+[source, yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-root-issuer
+  namespace: openshift-operators
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: root-ca
+  namespace: openshift-operators
+spec:
+  isCA: true
+  duration: 21600h # 900d
+  secretName: root-ca
+  commonName: root-ca.my-company.net
+  subject:
+    organizations:
+    - my-company.net
+  issuerRef:
+    name: selfsigned-root-issuer
+    kind: Issuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: root-ca
+spec:
+  ca:
+    secretName: root-ca
+----
++
+====
+[NOTE]
+The namespace of `selfsigned-root-issuer` issuer and `root-ca` certificate is `openshift-operators`. This is necessary because `root-ca` is a cluster issuer, so cert-manager will look for a referenced secret in its own namespace, which is `openshift-operators` in this case.
+====
+
+. Create an intermediate CA certificate for istiod:
```

Comment:
This section is irrelevant for istio-csr. Please take a look at these instructions once again: https://github.com/maistra/istio-operator/tree/maistra-2.4/deploy/examples/cert-manager/istio-csr

Reply:
I think I got myself confused, so I want to check. 

Do I replace theses steps:

Create an intermediate CA certificate for istiod:
Deploy SMCP:
Deploy apps:

with these:

Install cert-manager istio-csr:
Deploy Istio:
Deploy bookinfo app:

and then do i replace 

Test mTLS between apps: 

with

Test traffic:

i think i was looking at `custom-cert-rotation/README.md`file as the title is also "Integration with cert-manager and istio-csr", and didn't realize there was `istio-csr/README.md` so I want to check that I'm updating and adding the correct information. 

Which brings to another questions: do I need to create docs for `custom-cert-rotation` steps? Or are those no longer relevant?

Reply:
Steps:

1. Install the `cert-manager Operator for Red Hat OpenShift ` from the OCP cluster
2. Create the rootCA certificate on the namespace cert-manager


3. Create the istioCA certificate on the namespace Istio-system
4. Install Istio-CSR on the namespace Istio-system 
5. Deploy SMCP

Reply:
Steps:

Install the cert-manager Operator for Red Hat OpenShift  from the OCP cluster

Create the rootCA certificate on the namespace cert-manager

```
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: selfsigned-root-issuer
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: selfsigned-ca
spec:
  isCA: true
  duration: 21600h # 900d
  secretName: root-ca
  commonName: root-ca.my-company.net
  subject:
    organizations:
    - my-company.net
  issuerRef:
    name: selfsigned-root-issuer
    kind: Issuer
    group: cert-manager.io
---
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: root-ca
spec:
  ca:
    secretName: root-ca
```

Create the istioCA certificate on the namespace Istio-system

Install Istio-CSR on the namespace Istio-system

Deploy SMCP

Reply:
Steps:

1. Install the cert-manager Operator for Red Hat OpenShift on the  NS cert-manager-operator
2. Apply the rootCA on the NS `cert-manager`

```
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: selfsigned-root-issuer
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: selfsigned-ca
spec:
  isCA: true
  duration: 21600h # 900d
  secretName: root-ca
  commonName: root-ca.my-company.net
  subject:
    organizations:
    - my-company.net
  issuerRef:
    name: selfsigned-root-issuer
    kind: Issuer
    group: cert-manager.io
---
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: root-ca
spec:
  ca:
    secretName: root-ca

```
3. Apply the istioCA on the NS istio-system


```
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: istio-ca
spec:
  isCA: true
  duration: 21600h
  secretName: istio-ca
  commonName: istio-ca.my-company.net
  subject:
    organizations:
    - my-company.net
  issuerRef:
    name: root-ca
    kind: ClusterIssuer
    group: cert-manager.io
---
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: istio-ca
spec:
  ca:
    secretName: istio-ca
```

4. Install the istio-csr on `istio-system`

5. deploy the SMCP on `istio-system
`

```
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
metadata:
  name: basic
spec:
  security:
    certificateAuthority:
      cert-manager:
        address: cert-manager-istio-csr.%s.svc:443
      type: cert-manager
    dataPlane:
      mtls: true
    identity:
      type: ThirdParty
  tracing:
    type: None
  version: 2.4
```

then Verification

Reply:
cc @dymurray @jewertow :

Review these steps